core/build.rs

```diff
@@ -1,49 +0,0 @@
-// Copyright 2020 Parity Technologies (UK) Ltd.
-//
-// Permission is hereby granted, free of charge, to any person obtaining a
-// copy of this software and associated documentation files (the "Software"),
-// to deal in the Software without restriction, including without limitation
-// the rights to use, copy, modify, merge, publish, distribute, sublicense,
-// and/or sell copies of the Software, and to permit persons to whom the
-// Software is furnished to do so, subject to the following conditions:
-//
-// The above copyright notice and this permission notice shall be included in
-// all copies or substantial portions of the Software.
-//
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-// DEALINGS IN THE SOFTWARE.
-
-use pb_rs::{types::FileDescriptor, ConfigBuilder};
-use std::path::{Path, PathBuf};
-
-fn main() {
```

Comment:
We will need a CI check that ensures that the generated code is not modified!

Ideally, the CI check can use glob patterns so we auto-detect when new `.proto` files are added to the codebase.

Reply:
I'm not too familiar with Github actions so I have some questions. Do we want the build to fail when these files are modified? Is there a way to turn these off when we legitimately need to update these files? Also, I think when someone adds a new generated file and .proto file, the build should not fail?

Another option is to make a warning comment in the PR when proto/* files are added or modified. Another simpler approach which I've seen some projects do ([mitmproxy](https://github.com/mitmproxy/mitmproxy/tree/main/mitmproxy/contrib/kaitaistruct) for example) is to add these `proto/` directories to `.gitignore`. 

Thoughts?

Possible solutions:
- https://www.meziantou.net/executing-github-actions-jobs-or-steps-only-when-specific-files-change.htm
- https://github.com/marketplace/actions/comment-pull-request